xgcm/padding.py

```diff
@@ -0,0 +1,295 @@
+"""
+Handle all padding.
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Dict, Optional, Tuple, Union
+
+import xarray as xr
+
+if TYPE_CHECKING:
+    from .grid import Grid
+
+_XGCM_BOUNDARY_KWARG_TO_XARRAY_PAD_KWARG = {
+    "periodic": "wrap",
+    "fill": "constant",
+    "extend": "edge",
+    None: "wrap",  # current default is periodic. This should achieve that.
+}
+
+
+def _maybe_swap_dimension_names(da, from_name, to_name):
+    # renames 1D slices and swaps dimension names for higher dimensional slices
+    if to_name in da.dims:
+        da = da.rename({to_name: to_name + "dummy"})
+        if from_name in da.dims:
+            da = da.rename({from_name: to_name})
+        da = da.rename({to_name + "dummy": from_name})
+    else:
+        da = da.rename({from_name: to_name})
+    return da
+
+
+def _pad_face_connections(da, grid, padding_width, **kwargs):
```

Comment:
This function deserves its own docstring

Reply:
Good point! It is quite a beast at this point. Ill see if I can refactor it more once this matures.